src/HTSPDemuxer.cpp

```diff
@@ -226,6 +226,47 @@ PVR_ERROR CHTSPDemuxer::CurrentSignal ( PVR_SIGNAL_STATUS &sig )
   return PVR_ERROR_NO_ERROR;
 }
 
+int64_t CHTSPDemuxer::GetTimeshiftTime() const
```

Comment:
what's the problem with locking the mutex once on entry of CHTSPDemuxer::IsRealTimeStream()? Is it to expensive to reacquire the already locked mutex when entering GetTimeshiftTime() and GetTimeshiftBufferEnd()?